### ISSUE TITLE
Describe how to run tslint with type checking form the command line

### ISSUE THREAD

**Issue by scharf** [palantir/tslint#3014]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: CLI 

How to run `tslint` with type checking? I simply don't understand the documentation in [docs/usage/type-checking/index.md](https://github.com/palantir/tslint/blob/e22bf9e16549b070435cf903563352cba15ac2cb/docs/usage/type-checking/index.md)

- What is this program snippet shown in the docu used for?
- Where to put it?
- How run `tslint` with `--project` form the command line?

When I run it
```sh
tslint --project admin/tslint.json "project/**/*.ts{,x}"
```

I get
```
Invalid source file: project/main.ts. Ensure that the files supplied to lint 
have a .ts, .tsx, .d.ts, .js or .jsx extension.
```

**Comment by ajafff** [comment]
`--project` needs to point to your tsconfig.json. 
Try `tslint --project admin/tsconfig.json`
You also don't need to specify the files to lint. It will automatically include all files in the project. 

Note to myself: we definitely need to report errors when loading tsconfig.json fails.

**Comment by scharf** [comment]
But how does`tslint` know which files belong to the *project*?

The documentation of `--project` says:

>    The location of a tsconfig.json file that will be used to determine which files will be linted.

What is a `tsling` **project*? How to create a *project*?  I [cannot find anything](https://www.google.de/search?client=safari&rls=en&q=site:https://palantir.github.io/tslint/+project) talking relevant about the concept of `tslint` *projects*.

**Comment by scharf** [comment]
Ah! Looking into files of PR #2688 I understand that by project you mean the `tsconfig.json` file!

**Comment by scharf** [comment]
I am slightly dyslexic and I confused `tslint.json` with `tsconfig.json`! Maybe a full example would help 

```
tslint --config admin/tslint.json --project proj/tsconfig.json 
```

**Comment by adidahiya** [comment]
@scharf that's fair; a full example of using `--project` would certainly be helpful in the documentation.

**Comment by scharf** [comment]
in particular because the `--config` flag does not take the file that contains the string `config` (`tsconfig.json`) which causes a kind of  [Stroop effect](https://en.wikipedia.org/wiki/Stroop_effect).

 Say the COLOR, not the word:
![stroop_effect](https://user-images.githubusercontent.com/3731059/27994466-4fe54154-64bf-11e7-898e-b27b4fb2eb54.jpg)
